src/data/reports/glass-admin/GLASSDataMaintenanceDefaultRepository.ts

```diff
@@ -0,0 +1,195 @@
+            }))
+            .filter(module => module.name !== earModule);
+    }
+
+    private async getUploads(namespace: string): Promise<GLASSDataMaintenanceItem[]> {
+        const uploads = (await this.globalStorageClient.getObject<GLASSDataMaintenanceItem[]>(namespace)) ?? [];
+
+        return uploads;
+    }
+
+    private async deleteFileResources(items: Id[]): Promise<void> {
+        _.forEach(items, async item => {
+            try {
+                await this.api.delete(`/documents/${item}`).getData();
+            } catch (error) {
+                console.debug("File does not exist");
+            }
+        });
+    }
+
+    private getFilteredFiles(files: GLASSDataMaintenanceItem[], orgUnits: NamedRef[], module: Module) {
+        const lastDataSubmissionYear = new Date().getFullYear() - 1; // last submission year is the previous year
+
+        return _(files)
+            .map(upload => ({
+                ...upload,
+                id: upload.fileId,
+                orgUnitName: orgUnits.find(ou => ou.id === upload.orgUnit)?.name ?? "",
+            }))
+            .filter(upload => {
+                const isIncomplete = upload.status !== "COMPLETED";
+                const isBeforeSubmissionYear = parseInt(upload.period, 10) < lastDataSubmissionYear;
+                const isInModule = upload.module === module;
+
+                return isIncomplete && isBeforeSubmissionYear && isInModule;
+            })
+            .value();
+    }
+
+    private updateStatusById(
+        ids: string[],
+        uploads: GLASSDataMaintenanceItem[],
+        status: Status
+    ): GLASSDataMaintenanceItem[] {
+        return _.map(uploads, upload => {
+            if (_.includes(ids, upload.fileId)) {
+                return _.assign({}, upload, { status: status });
+            }
+            return upload;
+        });
+    }
+
+    private getRowIds(rows: GLASSDataMaintenanceItem[]): string[] {
+        return _(rows)
+            .filter(row => row.status !== "DELETED")
+            .map(row => row.id)
+            .value();
+    }
+
+    private getPager<T>(rows: T[], paging: Paging): Pager {
+        return {
+            page: paging.page,
+            pageSize: paging.pageSize,
+            pageCount: Math.ceil(rows.length / paging.pageSize),
+            total: rows.length,
+        };
+    }
+
+    private getPaginatedObjects<T>(rows: T[], sorting: Sorting<T>, paging: Paging): T[] {
+        return _(rows)
+            .orderBy([row => row[sorting.field]], [sorting.direction])
+            .drop((paging.page - 1) * paging.pageSize)
+            .take(paging.pageSize)
+            .value();
+    }
+
+    private paginate<T>(objects: T[], sorting: Sorting<T>, paging: Paging) {
+        const pager = this.getPager(objects, paging);
+        const paginatedObjects = this.getPaginatedObjects(objects, sorting, paging);
+
+        return { pager, objects: paginatedObjects };
+    }
+}
+
+const emptyPage = {
```

Comment:
When definining literal objects, be explicit with the type so TS can help you: `: GLASSMaintenancePaginatedObjects<GLASSDataMaintenanceItem>`